mm2src/mm2_p2p/src/behaviours/atomicdex.rs

```diff
@@ -824,11 +926,17 @@ fn maintain_connection_to_relays(swarm: &mut AtomicDexSwarm, bootstrap_addresses
                 if swarm.behaviour().core.gossipsub.is_connected_to_addr(&addr) {
                     continue;
                 }
+
+                if !pre_dial_check(&mut recently_dialed_peers, &addr) {
```

Comment:
and here we check the dialing one more time, if the peer had one address and it wasn't dialed yet, we would have marked it as dialed in the filer we did above and we won't be able to connect to it here.

we should also not update the dial expiry if we didn't dial (due to dial check failing)!

Reply:
> and here we check the dialing one more time

Was leftover, thanks.

> we should also not update the dial expiry if we didn't dial (due to dial check failing)!

I don't understand what you mean, but if you are saying that we shouldn't add peers to recently dialed peers I think that's exactly what we should do. If we fail to connect some peer, we shouldn't try again for sometime.

Reply:
> I don't understand what you mean, but if you are saying that we shouldn't add peers to recently dialed peers I think that's exactly what we should do. If we fail to connect some peer, we shouldn't try again for sometime.

e.g. for clarification:
we connected to some node, say bootstrap one, marked it as dialed with some expiry, then disconnected for some reason (you name it). Now when we try to connect back to maintain the min_connected, we will find that we dialed this node recently, but instead of just going with our day we would re-mark it is dialed again (even though we didn't dial it) and reset the expiry timer. in a bad situation where we can't find enough peers, the node will just be stuck there forever and ever resetting the timer.

Reply:
We don't need to reset the expiration time as that would kill its purpose (if the expiration time is 10 seconds, the peer should be available on the 11th second). And as you mentioned, peers could get stuck there indefinitely in certain situations.

Reply:
> We don't need to reset the expiration time as that would kill its purpose, peers could get stuck there indefinitely in certain situations.

but that's what we do now. we `check_and_mark_dialed` (**`and_mark_dialed`**) even if we didn't dial but we were just pre-checking, thus we set back the timeout one more time for a node that is already timed out, extending the timeout ever more.

Reply:
Misunderstood, sorry -- Yeah, I should remove the 2nd call.